### PR TITLE
[crossguid] Fix compilation on gcc v13

### LINKED ISSUE #31834: [crossguid] Build error on Linux with gcc v13
It seems that `<cstdint>` was implicitly included in previous versions of gcc but now compilation of crossguid fails like so:

```
[1/3] /usr/lib/ccache/bin/c++ -DGUID_LIBUUID -I/home/alex/code/vcpkg/buildtrees/crossguid/src/08ee1b7681-f70a1f9530.clean/include -I/home/alex/code/vcpkg/installed/x64-linux/include -fPIC -g -std=c++17 -fPIC -MD -MT CMakeFiles/crossguid.dir/src/guid.cpp.o -MF CMakeFiles/crossguid.dir/src/guid.cpp.o.d -o CMakeFiles/crossguid.dir/src/guid.cpp.o -c /home/alex/code/vcpkg/buildtrees/crossguid/src/08ee1b7681-f70a1f9530.clean/src/guid.cpp
FAILED: CMakeFiles/crossguid.dir/src/guid.cpp.o 
/usr/lib/ccache/bin/c++ -DGUID_LIBUUID -I/home/alex/code/vcpkg/buildtrees/crossguid/src/08ee1b7681-f70a1f9530.clean/include -I/home/alex/code/vcpkg/installed/x64-linux/include -fPIC -g -std=c++17 -fPIC -MD -MT CMakeFiles/crossguid.dir/src/guid.cpp.o -MF CMakeFiles/crossguid.dir/src/guid.cpp.o.d -o CMakeFiles/crossguid.dir/src/guid.cpp.o -c /home/alex/code/vcpkg/buildtrees/crossguid/src/08ee1b7681-f70a1f9530.clean/src/guid.cpp
In file included from /home/alex/code/vcpkg/buildtrees/crossguid/src/08ee1b7681-f70a1f9530.clean/src/guid.cpp:26:
/home/alex/code/vcpkg/buildtrees/crossguid/src/08ee1b7681-f70a1f9530.clean/include/crossguid/guid.hpp: In member function ‘std::size_t std::hash<xg::Guid>::operator()(const xg::Guid&) const’:
/home/alex/code/vcpkg/buildtrees/crossguid/src/08ee1b7681-f70a1f9530.clean/include/crossguid/guid.hpp:145:31: error: ‘uint64_t’ does not name a type
  145 |                         const uint64_t* p = reinterpret_cast<const uint64_t*>(guid.bytes().data());
      |                               ^~~~~~~~
/home/alex/code/vcpkg/buildtrees/crossguid/src/08ee1b7681-f70a1f9530.clean/include/crossguid/guid.hpp:39:1: note: ‘uint64_t’ is defined in header ‘<cstdint>’; did you forget to ‘#include <cstdint>’?
   38 | #include <iomanip>
  +++ |+#include <cstdint>
   39 | 
/home/alex/code/vcpkg/buildtrees/crossguid/src/08ee1b7681-f70a1f9530.clean/include/crossguid/guid.hpp:146:50: error: ‘uint64_t’ was not declared in this scope
  146 |                         return xg::details::hash<uint64_t, uint64_t>{}(p[0], p[1]);
      |                                                  ^~~~~~~~
/home/alex/code/vcpkg/buildtrees/crossguid/src/08ee1b7681-f70a1f9530.clean/include/crossguid/guid.hpp:146:50: note: ‘uint64_t’ is defined in header ‘<cstdint>’; did you forget to ‘#include <cstdint>’?
/home/alex/code/vcpkg/buildtrees/crossguid/src/08ee1b7681-f70a1f9530.clean/include/crossguid/guid.hpp:146:68: error: template argument 1 is invalid
  146 |                         return xg::details::hash<uint64_t, uint64_t>{}(p[0], p[1]);
      |                                                                    ^
/home/alex/code/vcpkg/buildtrees/crossguid/src/08ee1b7681-f70a1f9530.clean/include/crossguid/guid.hpp:146:72: error: ‘p’ was not declared in this scope
  146 |                         return xg::details::hash<uint64_t, uint64_t>{}(p[0], p[1]);
      |                                                                        ^
ninja: build stopped: subcommand failed.
```

### PR DESCRIPTION
**Pull request by alexdewar**
Crossguid no longer compiles with gcc v13 (tested on Linux), due to a missing include (cstdint). There is an upstream pull request to fix the issue which has been open for some time but not yet merged.

Let's patch the sources in vcpkg until upstream fixes it.

Fixes #31834.

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by alexdewar** [comment]
@microsoft-github-policy-service agree company="Imperial College London"

File: ports/crossguid/missing-include-cstdint.patch [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+gcc 13 moved some includes around and as a result <cstdint> is no longer transitively
+included [1]. Explicitly include it for uint{32,64}_t.
+
+[1] https://gcc.gnu.org/gcc-13/porting_to.html#header-dep-changes
+
+Signed-off-by: Khem Raj <raj.khem@gmail.com>
+---
+ include/crossguid/guid.hpp | 1 +
+ 1 file changed, 1 insertion(+)
+
```
Comment (Adela0814) [inline ports/crossguid/missing-include-cstdint.patch:15]: Please remove these useless codes, thanks.

**Review by Adela0814 — CHANGES_REQUESTED** [review]

**Comment by Adela0814** [comment]
Note: I will be converting your PR to draft status. When you respond, please revert to "ready for review". That way, I can be aware that you've responded since you can't modify the tags.

File: ports/crossguid/missing-include-cstdint.patch (hunk 1 above)
Comment (alexdewar) [reply ports/crossguid/missing-include-cstdint.patch]: Done

**Review by Adela0814 — APPROVED** [review]

**Pull request merged**